datalab/datalab_session/data_operations/data_operation.py

```diff
@@ -88,3 +95,64 @@ def set_output(self, output_data: dict):
 
     def get_output(self) -> dict:
         return cache.get(f'operation_{self.cache_key}_output')
+
+    def create_add_thumbnails_to_bucket(self, hdu_list: fits.HDUList, percent=None, cur_percent=None) -> list:
```

Comment:
This adds the fits file itself too, not just the thumbnails. So maybe a better name would be `create_and_store_fits` - I think the fits is the main data product and the thumbnails just come along with that.